hometask-1-OOP/src/Triangle.ts

```diff
@@ -1,3 +1,51 @@
+import { Point } from './Point';
 import { Shape } from './Shape';
 
-export class Triangle extends Shape {}
+export class Triangle extends Shape {
+	public constructor(point1: Point, point2: Point, point3: Point);
+	public constructor(point1: Point, point2: Point, point3: Point, color: string, filled: boolean);
+
+	public constructor(point1: Point, point2: Point, point3: Point, color?: string, filled?: boolean) {
+		const points = [point1, point2, point3];
+
+		if (color && typeof filled === 'boolean') {
+			super(points, color, filled);
+			return;
+		}
+
+		super(points);
+	}
+
+	public toString(): any {
+		const pointsStringified = this.points.map((point: Point, idx: number): string => `v${idx + 1}=${point.toString()}`).join(',');
```

Comment:
you can move this logic to a private method.

Reply:
Thanks!